narciuch/assignment2/test/TreeAncestorsTest.java

```diff
@@ -0,0 +1,96 @@
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static junit.framework.TestCase.assertTrue;
+
+public class TreeAncestorsTest {
+  TreeAncestors ancestorFinder = new TreeAncestors();
+  Node seven = new Node(7);
+
+  private Node getTestTree() {
+    Node three = new Node(3);
+    Node four = new Node(4);
+    Node two = new Node(2);
+    Node five = new Node(5);
+    Node eight = new Node(8);
+    Node one = new Node(1);
+    Node six = new Node(6);
+    seven.setLeft(three);
+    seven.setRight(four);
+    four.setRight(eight);
+    three.setLeft(two);
+    three.setRight(five);
+    two.setLeft(one);
+    two.setRight(six);
+    return seven;
+  }
+
+  @Test
+  public void noAncestorsInEmptyTree() {
+    assertTrue(ancestorFinder.getAncestors(7, null).size() == 0);
+  }
+
+  @Test
+  public void noAncestorsInOneElemTree() {
+    assertTrue(ancestorFinder.getAncestors(7, new Node(65)).size() == 0);
+    assertTrue(ancestorFinder.getAncestors(7, new Node(7)).size() == 0);
+  }
+
+  @Test
+  public void noAncestorsIfKeyNotInTree() {
+    assertTrue(ancestorFinder.getAncestors(44, getTestTree()).size() == 0);
+  }
+
+  @Test
+  public void correctlyIdentifiesAncestors() {
+    Node three = new Node(3);
+    Node four = new Node(4);
+    Node two = new Node(2);
+    Node five = new Node(5);
+    Node eight = new Node(8);
+    Node one = new Node(1);
+    Node six = new Node(6);
+    seven.setLeft(three);
```

Comment:
This way of building tree seems quite inconvenient. It would be much easier if `Node` provided methods like:
```
/** Adds node with given value as left child of current node and returns newly created child. */
public Node<T> addLeftChild(T value) {...}
```

Reply:
Yes, this is a much nicer solution.